src/zarr/core/group.py

```diff
@@ -987,22 +987,24 @@ async def create_array(
         self,
         name: str,
         *,
-        shape: ShapeLike,
-        dtype: npt.DTypeLike,
+        shape: ShapeLike | None = None,
+        dtype: npt.DTypeLike | None = None,
+        data: np.ndarray[Any, np.dtype[Any]] | None = None,
         chunks: ChunkCoords | Literal["auto"] = "auto",
         shards: ShardsLike | None = None,
         filters: FiltersLike = "auto",
         compressors: CompressorsLike = "auto",
         compressor: CompressorLike = "auto",
         serializer: SerializerLike = "auto",
-        fill_value: Any | None = 0,
+        fill_value: Any | None = None,
```

Comment:
Does this mean the default fill value is no longer `0`?

Reply:
in `main` the default fill value is not consistent. in some places it's 0 (e.g., here), and in [other places](https://github.com/zarr-developers/zarr-python/blob/64b9a370a94c8d8afcfd06f93ca25d77e668cdd6/src/zarr/core/array.py#L4019) it's `None`. This PR makes it consistently `None`.

Reply:
I think worth shouting loudly that this has changed (for certain functions) in a changelog entry then, if it changes the default behaviour.

Reply:
let me know what you think of the language I added to the changelog in my recent commit